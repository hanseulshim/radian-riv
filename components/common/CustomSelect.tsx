import Select from 'react-select'

interface Option {
  label: string
  value: any
}

interface Props {
  options: Option[]
  value: Option
  onChange: (option: Option, key: string) => void
  label?: string
  placeholder?: string
  className?: string
  disabled?: boolean
}

const customStyles = {
  option: provided => ({
    ...provided,
    cursor: 'pointer',
    color: 'black',
    backgroundColor: '#ffffff',
    '&:hover': {
      backgroundColor: 'rgba(61, 57, 53, 0.05)'
    }
  }),
  menu: provided => ({
    ...provided,
    zIndex: '3'
  }),
  control: provided => ({
    ...provided,
    cursor: 'pointer',
    border: 'none',
    borderBottom: '1px solid #00bab3',
    boxShadow: 'none',
    '&:hover': {
      borderBottom: '1px solid #00bab3'
    },
    borderRadius: 0,
    width: '100%'
  })
}

export default function CustomSelect({
  options,
  value,
  onChange,
  label,
  placeholder,
  className,
  disabled
}: Props) {
  return (
    <div className="custom-select-container">
      {value && <div className="custom-select-label">{label}</div>}
      <Select
        options={options}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        className={className}
        isDisabled={disabled}
        styles={customStyles}
      />
    </div>
  )
}
