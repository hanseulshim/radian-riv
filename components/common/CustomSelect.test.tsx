import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import CustomSelect from './CustomSelect'

describe('CustomSelect', () => {
  let component
  beforeAll(() => {
    const props = {
      options: [{ label: 'test', value: 'test' }],
      value: { label: 'test', value: 'test' },
      label: 'test',
      onChange: () => {}
    }
    component = shallow(<CustomSelect {...props} />)
  })
  it('renders container', () => {
    expect(component.find('.custom-select-container')).toHaveLength(1)
  })
  it('has label component', () => {
    expect(component.find('.custom-select-label').text()).toEqual('test')
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
