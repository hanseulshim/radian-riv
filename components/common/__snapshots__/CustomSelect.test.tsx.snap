// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect should match snapshot 1`] = `
<div
  className="custom-select-container"
>
  <div
    className="custom-select-label"
  >
    test
  </div>
  <StateManager
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    onChange={[Function]}
    options={
      Array [
        Object {
          "label": "test",
          "value": "test",
        },
      ]
    }
    styles={
      Object {
        "control": [Function],
        "menu": [Function],
        "option": [Function],
      }
    }
    value={
      Object {
        "label": "test",
        "value": "test",
      }
    }
  />
</div>
`;
